File: assets/codebase/rendering/AxisTestZSpinComponent.cs
namespace gameplay.rendering {
    /// <summary>
    /// Rotates the parent entity around its local Z axis using deterministic absolute runtime time.
    /// </summary>
    public sealed class AxisTestZSpinComponent : UpdateComponent {
        /// <summary>
        /// Gets or sets the base roll offset in radians applied before time-based rotation.
        /// </summary>
        public float BaseRollRadians { get; set; }

        /// <summary>
        /// Gets or sets the angular speed in radians per second.
        /// </summary>
        public float AngularSpeedRadians { get; set; }

        /// <summary>
        /// Evaluates the current orientation from total elapsed runtime time.
        /// </summary>
        public override void Update() {
            base.Update();

            double rollRadians = BaseRollRadians + (AngularSpeedRadians * Core.Instance.TotalElapsedSeconds);
            float4 orientation;
            float4.CreateFromYawPitchRoll(0f, 0f, (float)rollRadians, out orientation);
            orientation.Normalize();
            Parent.LocalOrientation = orientation;
        }
    }
}

File: windows-build-20260529-stack-box-direct/cooked/code/gameplay/AxisTestZSpinComponent.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "AxisTestZSpinComponent.hpp"
#include "UpdateComponent.hpp"
#include "float4.hpp"
#include "Core.hpp"

AxisTestZSpinComponent::AxisTestZSpinComponent() : AngularSpeedRadians(), BaseRollRadians()
{
}

float AxisTestZSpinComponent::get_AngularSpeedRadians()
{
return this->AngularSpeedRadians;
}

void AxisTestZSpinComponent::set_AngularSpeedRadians(float value)
{
this->AngularSpeedRadians = value;
}

float AxisTestZSpinComponent::get_BaseRollRadians()
{
return this->BaseRollRadians;
}

void AxisTestZSpinComponent::set_BaseRollRadians(float value)
{
this->BaseRollRadians = value;
}

void AxisTestZSpinComponent::Update()
{
UpdateComponent::Update();
const double rollRadians = this->BaseRollRadians + (this->AngularSpeedRadians * Core::get_Instance()->get_TotalElapsedSeconds());
float4 orientation;
float4::CreateFromYawPitchRoll(0.0f, 0.0f, static_cast<float>(rollRadians), orientation);
orientation.Normalize();
Parent->set_LocalOrientation(orientation);
}

